src/API/Middlewares/RequestValidationMiddleware.ts

```diff
@@ -0,0 +1,27 @@
+import { IllegalArgumentException } from '@po.et/poet-js'
+import * as Joi from 'joi'
+interface RequestValidationParams {
+  readonly query?: (values: object) => {}
```

Comment:
If `values` isn't an array, rename to `value`. I'm also thinking something like `querySchema`, `bodySchema` and `paramsSchema` would be more descriptive.

Reply:
it's an object and match with `ctx.query`, `ctx.params` and `ctx.request.body`.

What do you think if we change the name to `queryRequestValues`, for example?

Reply:
I meant renaming `values` (the parameter of the function), not the keys of `RequestValidationParams`. Do you want to rename the keys, too?

Reply:
for example:

```
interface RequestValidationParams {
  readonly query?: (queryRequestValues: object) => {}
  readonly body?: (bodyRequestValues: object) => {}
  readonly params?: (paramsRequestValues: object) => {}
  readonly options?: object
}
```